src/main/java/com/suchocki/parkingmeter/dao/DriverPaymentDAOImpl.java

```diff
@@ -0,0 +1,62 @@
+package com.suchocki.parkingmeter.dao;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import org.springframework.stereotype.Repository;
+import com.suchocki.parkingmeter.database.FakeDatabaseStub;
+import com.suchocki.parkingmeter.entity.DriverPayment;
+
+@Repository
+public class DriverPaymentDAOImpl implements DriverPaymentDAO {
+
+	@Override
+	public void save(DriverPayment driverPayment) {
+		driverPayment.setPayDate(new Date());
+		driverPayment.setId(FakeDatabaseStub.driverPayments.size() + 1);
```

Comment:
czy to jest bezpieczne w wielowątkowym środowisku?

Reply:
Nie, przez użycie klasy `java.util.Date`. Lepiej byłoby używać `java.time.LocalDate` z Javy 8

Reply:
Nie, właściwie cały ten pomysł na nadawanie id na podstawie wielkości listy jest zły. Tutaj dwa wątki mogą jednocześnie dodawać. Przy usuwaniu elementów, po operacji usunięcia, dwa mogą mieć takie samo id. Można by to rozwiązać przez statyczny licznik i synchronizację go. Nie przejmowałem się tym, bo baza miała być "oszukana", więc założyłem, że mogę pominąć takie problemy. Przy zapisywaniu do prawdziwej bazy, to ona ustalałaby id (autoincrement).